src/details/ArborX_DetailsUtils.hpp

```diff
@@ -526,9 +526,21 @@ void reallocWithoutInitializing(View &v,
                                 size_t n6 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
                                 size_t n7 = KOKKOS_IMPL_CTOR_DEFAULT_ARG)
 {
-  static_assert(View::is_managed, "Can only realloc managed views");
-  v = View(Kokkos::view_alloc(Kokkos::WithoutInitializing, v.label()), n0, n1,
-           n2, n3, n4, n5, n6, n7);
+  size_t ranks[8] = {n0, n1, n2, n3, n4, n5, n6, n7};
+  bool allocation_necessary = false;
+  for (unsigned int dim = 0; dim < v.rank_dynamic; ++dim)
+    if (ranks[dim] != v.extent(dim))
+    {
+      allocation_necessary = true;
+      break;
+    }
+
+  if (allocation_necessary)
+  {
+    static_assert(View::is_managed, "Can only realloc managed views");
```

Comment:
I don't think it makes sense to put this `static_assert` inside the `if` clause as it is always applicable.

Reply:
It doesn't make any difference. It's compiled anyway and the check runs independently of which branch we take at run-time.